src/java-interop/java-interop-logger.c

```diff
@@ -94,21 +94,37 @@ open_file (LogCategories category, const char *path, const char *override_dir, c
 	return f;
 }
 
+static const char *gref_file = NULL;
+static const char *lref_file = NULL;
+static int light_gref  = 0;
+static int light_lref  = 0;
```

Comment:
Why do we have above variables as static instead of having them as local function variables as before?

Reply:
Because they are set in `init_logging_categories` when parsing the arguments but used in `init_reference_logging`. When this code is converted to C++ they will become instance variables.